jclouds-plugin/src/main/java/jenkins/plugins/jclouds/compute/JCloudsSlaveTemplate.java

```diff
@@ -312,41 +389,27 @@ public NodeMetadata get() {
         }
 
         Statement initStatement = null;
-        Statement bootstrap = null;
 
         if (this.preExistingJenkinsUser) {
             if (this.initScript.length() > 0) {
                 initStatement = Statements.exec(this.initScript);
             }
         } else {
-            // setup the jcloudTemplate to customize the nodeMetadata with jdk, etc. also opening ports
+            // provision jenkins user
             AdminAccess adminAccess = AdminAccess.builder().adminUsername(getJenkinsUser())
-                    .installAdminPrivateKey(installPrivateKey) // some VCS such as Git use SSH authentication
-                    .grantSudoToAdminUser(allowSudo) // no need
-                    .adminPrivateKey(getCloud().privateKey) // temporary due to jclouds bug
-                    .authorizeAdminPublicKey(true).adminPublicKey(getCloud().publicKey).adminHome(getFsRoot()).build();
-
+                .installAdminPrivateKey(installPrivateKey) // some VCS such as Git use SSH authentication
+                .grantSudoToAdminUser(allowSudo) // no need
+                .adminPrivateKey(getJenkinsPrivateKey()) // temporary due to jclouds bug
+                .authorizeAdminPublicKey(true).adminPublicKey(getJenkinsPublicKey()).adminHome(getFsRoot()).build();
```

Comment:
Actually, I do mind. After all, this is just a PR, *not* a release
request. And it is certainly not my last PR. So: Why don't you want
ongoing development merged? Otherwise, ongoing development im my repo
gets accumulated in this single PR making it more complex. This PR is
already complex enough.

Regarding tests:
I think, there is currently a *big* blocker with that (regarding a
serious classloader and/or dependency problem). See:

This commit from @mavlyutov :
https://github.com/jenkinsci/jclouds-plugin/commit/c6ca88f46a877750e08333934c82ad2a50796007#diff-318c2de5828f928365639960aadd2eec

And this post:
https://groups.google.com/d/msgid/jenkinsci-dev/9440328a-9558-45c8-9f85-f68b683da25a%40googlegroups.com?utm_medium=email&utm_source=footer

Cheers
 - Fritz

On 02/23/2015 06:01 PM, Oliver Gondža wrote:
> In
> jclouds-plugin/src/main/java/jenkins/plugins/jclouds/compute/JCloudsSlaveTemplate.java
> <https://github.com/jenkinsci/jclouds-plugin/pull/83#discussion_r25180506>:
> 
>>              AdminAccess adminAccess = AdminAccess.builder().adminUsername(getJenkinsUser())
>> -                    .installAdminPrivateKey(installPrivateKey) // some VCS such as Git use SSH authentication
>> -                    .grantSudoToAdminUser(allowSudo) // no need
>> -                    .adminPrivateKey(getCloud().privateKey) // temporary due to jclouds bug
>> -                    .authorizeAdminPublicKey(true).adminPublicKey(getCloud().publicKey).adminHome(getFsRoot()).build();
>> -
>> +                .installAdminPrivateKey(installPrivateKey) // some VCS such as Git use SSH authentication
>> +                .grantSudoToAdminUser(allowSudo) // no need
>> +                .adminPrivateKey(getJenkinsPrivateKey()) // temporary due to jclouds bug
>> +                .authorizeAdminPublicKey(true).adminPublicKey(getJenkinsPublicKey()).adminHome(getFsRoot()).build();
> 
> Yeah, I failed to understand what |AdminAccess| is supposed to do and
> how. Anyway, I wanted to create some end-to-end ATH
> <https://github.com/jenkinsci/acceptance-test-harness> tests for
> JClouds/OpenStack for quite some time. If you do not mind, I suggest not
> to merge this before we have some coverage.
> 
> All in all I like the change. Great work on preserving backward
> compatibility!
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/jclouds-plugin/pull/83/files#r25180506>.
>